src/tests/SFA.DAS.EmployerIncentives.Commands.UnitTests/Services/LearnerService/WhenRefreshCalledAndDataLocksAreChecked.cs

```diff
@@ -0,0 +1,125 @@
+﻿using AutoFixture;
+using FluentAssertions;
+using NUnit.Framework;
+using SFA.DAS.EmployerIncentives.Commands.Services.LearnerMatchApi;
+using SFA.DAS.EmployerIncentives.Domain.ApprenticeshipIncentives.ValueTypes;
+using SFA.DAS.EmployerIncentives.UnitTests.Shared;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Commands.UnitTests.Services.LearnerService
+{
+    //    Conditions for DataLock:
+    // 1. ZPROG001 reference
+    // 2. PriceEpisode.StartDate <= NexPayment.DueDate => PriceEpisode.EndDate
+    // 3. Period.Period == NexPayment.CollectionPeriod
+    // 4. Period.ApprenticeshipId == ApprenticeshipIncentive.ApprenticeshipId
+    // 5. Period.IsPayable == false
+    public class WhenRefreshCalledAndDataLocksAreChecked
```

Comment:
You probably want a test for each condition in HasProviderDataLocks rather than a single catch all test.